### DISCUSSION TITLE
Being able to update a specific document index and keep the other documents untouched

### DISCUSSION THREAD

**Discussion by othomann** [mikefarah/yq#1232] [General]
Hi,

I have a yaml file that contains multiple documents:

```
---
defaultChannel: alpha
description: ""
name: private-worker-agent
schema: olm.package
---
schema: olm.channel
package: private-worker-agent
name: alpha
entries:
  - name: private-worker-agent.v2.2.32
    replaces: private-worker-agent.v2.2.8
---
image: icr.io/continuous-delivery/pipeline/agent-operator-bundle:0.11.13-olm
name: private-worker-agent.v2.2.32
package: private-worker-agent
properties:
  - type: olm.gvk
    value:
....
```

I would like to update only the document at index 1 to add new entries. I tried:

```
olivier@MacBook-Pro catalog % yq4 'select(document_index == 1) | .entries += {"name": "private-worker-agent.v"+ env(VERSION), "replaces": "private-worker-agent.v2.2.32"}' operator.yaml     
```

This returns:
```
schema: olm.channel
package: private-worker-agent
name: alpha
entries:
  - name: private-worker-agent.v2.2.32
    replaces: private-worker-agent.v2.2.8
  - name: private-worker-agent.v2.2.33
    replaces: private-worker-agent.v2.2.32
```

But it doesn't output all the other document indexes. How is it possible to update the document at index 1 and just output all other documents as untouched.

Thanks for any help. I am using `yq 4.25.2`.

I tried with yq 2.4.1 using -d 1. Seems to work, but I cannot figure out how to insert a new entry with that version.

**Comment by mikefarah** [comment]
Hi,

This is just a order precedence problem, what you need to do is only select that document _for_ the update, as opposed to selecting the document and then updating it. 

TLDR: you need brackets

```bash
yq4 '(select(document_index == 1) | .entries) += {"name": "private-worker-agent.v"+ env(VERSION), "replaces": "private-worker-agent.v2.2.32"}' operator.yaml
```
